src/Nethermind/Nethermind.Consensus.Clique/CliqueSealer.cs

```diff
@@ -31,18 +31,16 @@ namespace Nethermind.Consensus.Clique
     public class CliqueSealer : ISealer
     {
         private readonly ILogger _logger;
-        private readonly Address _sealerAddress;
         private readonly ISnapshotManager _snapshotManager;
-        private readonly IBasicWallet _wallet;
-        private ICliqueConfig _config;
+        private readonly ISigner _signer;
+        private readonly ICliqueConfig _config;
 
-        public CliqueSealer(IBasicWallet wallet, ICliqueConfig config, ISnapshotManager snapshotManager, Address sealerAddress, ILogManager logManager)
+        public CliqueSealer(ISigner signer, ICliqueConfig config, ISnapshotManager snapshotManager, ILogManager logManager)
```

Comment:
and this is fine - it is fine to pass the signer to the sealer but they should not coexist outside
what you can do -> add an Address field to the Sealer interface -> it makes perfect sense as in every single block production scenario we have the Address of the sealer -> which becomes a block author